src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -77,14 +77,17 @@ public void addBinaryField(FieldInfo field, DocValuesProducer valuesProducer) th
     }
 
     public void addKNNBinaryField(FieldInfo field, DocValuesProducer valuesProducer) throws IOException {
-
         // Get values to be indexed
         BinaryDocValues values = valuesProducer.getBinary(field);
         KNNCodecUtil.Pair pair = KNNCodecUtil.getFloats(values);
         if (pair.vectors.length == 0 || pair.docs.length == 0) {
             logger.info("Skipping engine index creation as there are no vectors or docs in the documents");
             return;
         }
+        // for Lucene engine the creation of index is done by code delegate, no need in additional native data structures
+        if (isLuceneEngine(field)) {
```

Comment:
How about we move this check to caller  ```addBinaryField ``` and dont even come here if the Engine is ```Lucene```

```suggestion
        if (!isLuceneEngine(field) &&  field.attributes().containsKey(KNNVectorFieldMapper.KNN_FIELD)) {
             addKNNBinaryField(field, valuesProducer);
        }
```

Reply:
I would really like to see who is the caller of this function before we move this code to the caller so that we know if that make sense to add over there or not. From the first look it looks like we can add it in the caller. 

Also, is there any specific reason why we have added this after few lines of code in the  function: addKNNBinaryField

Reply:
The function is being called from the same class https://github.com/opensearch-project/k-NN/blob/feature/lucene-knn/src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java#L75, existing logic is to add the binary field and then add native index part if the field is knn. 
We can refactor a bit more and pull the check high in the call stack to the caller, as you've mentioned. It will have some code duplication but I guess that's fins in this case

Reply:
> 

Agree, let me refactor this part